Fix param.Action (#9).

ceball opened the pull request.

ceball committed:
Fix param.Action (#9).

ceball commented on parambokeh/__init__.py:
```diff
@@ -174,6 +174,9 @@ def __call__(self, parameterized, doc=None, plots=[], **params):
         p_obj = self.parameterized.params(p_name)
+        if isinstance(p_obj, param.Action):
+            getattr(self.parameterized, p_name)(self.parameterized)
+            return
```
Other widgets won't be refreshed - is that a problem? What if the callback alters another parameter value? (I decided not to get into this now, deferring it to 'dependencies between parameters and code'...)

philippjfr replied:
>deferring it to 'dependencies between parameters and code'

+1

jbednar replied:
Sounds fine.

philippjfr commented on parambokeh/comms.py:
```diff
@@ -225,6 +225,10 @@ def _handle_msg(self, msg):
+            # TODO: isn't this cutting out info needed to understand what's gone wrong?
+            # Since it's only going to the js console, maybe we could just show everything
+            # (error = traceback.format_exc() or something like that)? Separately we do need a mechanism
+            # to report reasonable messages to users, though.
```
All true, if we can find a way to make tracebacks readable on the JS console I'd be very happy to change this (and port the change back to holoviews).

jbednar commented on examples/user_guide/Introduction.ipynb:
```diff
@@ -124,6 +124,22 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Example.timestamps records the times you pressed the \"record timestamp\" button."
```
Good example, thanks.

jbednar commented:
Happy to merge, but it would be nice if conda was working so the tests could complete.

ceball commented:
Seems like the same problem reported in https://github.com/conda/conda/issues/6468. However, the version of conda has not changed since the last successful build (4.3.30), so I'd guess something has changed at travis (e.g. related to docker, or the linux image, etc). I have not investigated.

ceball committed:
Test to see if new travis image is a problem.

ceball committed:
Build still fails even after reverting Travis image - revert the test.

ceball commented:
The problem appears to be intermittent :(

E.g. two jobs in the same build (same codebase, same image); one gets past the conda download/install stage, the other one fails: https://travis-ci.org/ioam/parambokeh/builds/317743629?utm_source=github_status&utm_medium=notification

ceball committed:
Avoid conda progress bar on travis to try to mitigate intermittent failures.

See https://github.com/conda/conda/issues/6468.

Also note that the existing duplication of commands and amount of bash inside a 'config' file is being tackled separately...

ceball committed:
Fix previous commit.

I was a victim of all the duplication even while writing about it...
